pkg/endpoints/pipeline.go

```diff
@@ -112,44 +139,63 @@ func (r Resource) getPipeline(request *restful.Request, response *restful.Respon
 	response.WriteEntity(pipeline)
 }
 
-/* Get all pipelines in a given namespace: the caller needs to handle any errors */
+/* Get all pipelines in a given namespace: the caller needs to handle any errors,
+an empty v1alpha1.Pipeline{} is returned if no pipeline is found */
 func (r Resource) getPipelineImpl(name, namespace string) (v1alpha1.Pipeline, error) {
 	logging.Log.Debugf("in getPipelineImpl, name %s, namespace %s", name, namespace)
 
 	pipelines := r.PipelineClient.TektonV1alpha1().Pipelines(namespace)
 	pipeline, err := pipelines.Get(name, metav1.GetOptions{})
 	if err != nil {
 		logging.Log.Errorf("could not retrieve the pipeline called %s in namespace %s", name, namespace)
-		return *pipeline, err
+		return v1alpha1.Pipeline{}, err
 	}
-	logging.Log.Debug("Found the pipeline definition OK")
+	logging.Log.Debugf("Found the pipeline definition OK")
 	return *pipeline, nil
 }
 
 /* Get all pipeline runs in a given namespace */
 func (r Resource) getAllPipelineRuns(request *restful.Request, response *restful.Response) {
 	namespace := request.PathParameter("namespace")
-	pipelineruns := r.PipelineClient.TektonV1alpha1().PipelineRuns(namespace)
 	repository := request.QueryParameter("repository")
 
 	logging.Log.Debugf("In getAllPipelineRuns: namespace: `%s`, repository query: `%s`", namespace, repository)
 
+	pipelinerunList, appError := r.GetAllPipelineRunsImpl(namespace, repository)
+	if appError.ERROR != nil {
+		logging.Log.Debugf("+%v", pipelinerunList.Items)
+		utils.RespondError(response, appError.ERROR, appError.CODE)
+	} else {
+		// no errors so return the pipeline run list
+		logging.Log.Debugf("+%v", pipelinerunList.Items)
+		response.WriteEntity(pipelinerunList)
+	}
+}
+
+// GetAllPipelineRunsImpl ...
+/* Returns a pointer to a list of PipelineRuns in a given namespace,
+optionally with a repository query as a parameter */
```

Comment:
Instead of saying the `repository` parameter is `optional`, please document in the comment that if the repository parameter is an empty string, then `GetAllPipelineRunsImpl` will return all the PipelineRuns in the given namespace.

There are no optional parameters in Go, so this change avoids some confusion and better explains the functionality.